### PR TITLE
Coverage tests for xTaskPriorityDisinherit, and vTaskPriorityDisinheritAftertimeout

### PR DESCRIPTION
**Pull request by adam-scislowicz**
Description
-----------
Coverage tests for xTaskPriorityDisinherit, and vTaskPriorityDisinheritAftertimeout

Test Steps
-----------
cd FreeRTOS/Test/CMock/smp
rm -rf ../build/generated
make lcovhtml

view the resulting generated file with a web browser: FreeRTOS/Test/CMock/build/generated/smp/coverage/Source/tasks.c.gcov.html

Related Issue
-----------
TS-24213, TS-24214

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c [hunk 1]
```diff
@@ -631,227 +645,613 @@ void test_task_delay_until_with_config_assert( void )
...
+ *          if( pxTCB->uxMutexesHeld == ( UBaseType_t ) 0 )
  * @endcode
- * As two tasks arecreated, this covers both branches of the above conditional
- * in addition to the function body.
+ *
+ * Cover the case where priority has changed due to inheritance
+ * because of a shared mutex.
+ *
  */
-void test_coverage_prvAddNewTaskToReadyList_create_two_tasks_with_the_first_suspended( void )
```
Comment (chinglee-iot) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c:646]: It looks like some of the test cases are removed in this PR?
test_coverage_prvAddNewTaskToReadyList_create_two_tasks_with_the_first_suspended
test_coverage_prvAddNewTaskToReadyList_create_more_idle_tasks_than_cores
test_coverage_vTaskCoreAffinitySet_task_core_affinity_set_task_implied
test_coverage_vTaskCoreAffinitySet_task_core_affinity_set_task_explicit
....
Why are we removing existing test cases?

File: FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c [hunk 2]
```diff
@@ -631,227 +645,613 @@ void test_task_delay_until_with_config_assert( void )
...
+void test_coverage_xTaskPriorityDisinherit_current_is_same( void )
```
Comment (chinglee-iot) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c:662]: There are existing test cases for single core.
Why are we cover this function in coverage test?

Comment (chinglee-iot) [reply FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c:662]: Attached image is the function coverage before this PR.
image

**Comment by adam-scislowicz** [comment]
closing for the same reason that PR8 was closed.

**Pull request closed without merging**